src/HttpMethodsClient.php

```diff
@@ -17,11 +18,14 @@ final class HttpMethodsClient implements HttpMethodsClientInterface
     private $httpClient;
 
     /**
-     * @var RequestFactory
+     * @var RequestFactory|RequestFactoryInterface
      */
     private $requestFactory;
 
-    public function __construct(ClientInterface $httpClient, RequestFactory $requestFactory)
+    /**
+     * @param RequestFactory|RequestFactoryInterface
+     */
+    public function __construct(ClientInterface $httpClient, $requestFactory)
     {
```

Comment:
i would love to have a type check here:

```
if (!$requestFactory instanceof RequestFactory && !$requestFactory instanceof RequestFactoryInterface) {
    throw new \InvalidArgumentException(sprintf('Factory must be of type %s or %s, %s given', RequestFactory::class, RequestFactoryInterface::class, is_object($requestFactory) ? get_class($requestFactory) : gettype($requestFactory)));
}
```

Reply:
I've added a check more like what PHP 8 would do if:

<img width="1034" alt="image" src="https://user-images.githubusercontent.com/2829600/86228683-9222fe00-bb86-11ea-961d-0539f7cb639e.png">

Reply:
awesome, thanks!